src/hooks/useActiveDescendantListFocus/__tests__/useActiveDescendantListFocus.jest.js

```diff
@@ -78,37 +141,144 @@ function runListUnitTest(isHorizontal) {
+
+    expect(result.current.visualFocusItemIndex).toEqual(undefined);
   });
 
   it("should not navigate to next item when user try to navigate by using keys for the  opposite dimension to the list dimension ", async () => {
     const onItemClick = jest.fn();
-    const { result } = renderHookForTest({ onItemClick, isHorizontal });
+    const { result } = renderHookForTest({ onItemClick, isHorizontal, defaultVisualFocusFirstIndex });
 
     act(() => {
       // set focus on the list's element which in charge on natural focus element
       element.focus();
+      // move visual focus to first item
+      userEvent.keyboard(oppositeMoveForwardKey);
+    });
 
+    let before = result.current.visualFocusItemIndex;
+    act(() => {
       // move visual focus to first item
       userEvent.keyboard(oppositeMoveForwardKey);
     });
 
-    expect(result.current.visualFocusItemIndex === 0).toBeFalsy();
+    expect(result.current.visualFocusItemIndex).toEqual(before);
+  });
+  it("should change visually focused item when calling setVisualFocusItemId with triggeredWithKeyboard argument", async () => {
+    const { result } = renderHookForTest({ isHorizontal, defaultVisualFocusFirstIndex });
+
+    act(() => {
+      // set focus on the list's element which in charge on natural focus element
+      element.focus();
+    });
+
+    act(() => {
+      const isTriggeredByKeyboard = true;
+      // Change visually focused item by using setVisualFocusItemId function
+      result.current.setVisualFocusItemId(THIRD_ITEM_ID, isTriggeredByKeyboard);
+    });
+
+    // Now the visual focus item index should be the index of THIRD_ITEM_ID item
+    expect(result.current.visualFocusItemIndex).toEqual(3);
   });
 }
 
+const combineFeatures = features => {
+  const combinations = [];
+  // all closed
+  const allClosedComb = {};
+  for (let feature of features) {
+    allClosedComb[feature] = false;
+  }
+  combinations.push(allClosedComb);
+
+  // all open
+  const allOpenComb = {};
+  for (let feature of features) {
+    allOpenComb[feature] = true;
+  }
+  combinations.push(allOpenComb);
+
+  // one open all other closed
+  for (const feature of features) {
+    const oneOpen = {};
+    oneOpen[feature] = true;
+    for (const otherFeature of features) {
+      oneOpen[otherFeature] = false;
+    }
+    combinations.push(oneOpen);
+  }
+
+  return combinations;
+};
+
 describe("useActiveDescendantListFocus", () => {
   afterEach(() => {
     element.remove();
     cleanup();
   });
 
-  describe("Vertical list", () => {
-    const isHorizontal = false;
-    runListUnitTest(isHorizontal);
+  const features = {
+    isHorizontal: { yes: "Horizontal", no: "Vertical" },
+    defaultVisualFocusFirstIndex: { yes: "KeepOpen", no: "" }
+  };
+
+  const featureCombinations = combineFeatures(Object.keys(features));
+  for (const featureComb of featureCombinations) {
+    describe(`${Object.entries(featureComb)
```

Comment:
consider extracting the calculation of the suite name to a separate variable. 
While this is ok, it's unconventional, and will probably be overwhelming. It's best being as explicit as possible

Reply:
will do in a different pr as we discuss